jujugui/static/gui/src/app/components/user-profile/agreement-list.js

```diff
@@ -0,0 +1,167 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('agreement-list', function() {
+
+  juju.components.AgreementList = React.createClass({
+    xhrs: [],
```

Comment:
This should be defined in the `getInitialState` method as `this.activeXhr = []` as right now it's shared by all instances of `AgreementList`. Which isn't likely an issue, but a good practice to get into.